tsdb/functions.go

```diff
@@ -66,20 +67,35 @@ func initializeMapFunc(c *influxql.Call) (mapFunc, error) {
 	case "median":
 		return MapStddev, nil
 	case "min":
-		return MapMin, nil
+		return func(itr Iterator) interface{} {
+			return MapMin(itr, c.Fields()[0])
+		}, nil
 	case "max":
-		return MapMax, nil
+		return func(itr Iterator) interface{} {
+			return MapMax(itr, c.Fields()[0])
+		}, nil
 	case "spread":
 		return MapSpread, nil
 	case "stddev":
 		return MapStddev, nil
 	case "first":
-		return MapFirst, nil
+		return func(itr Iterator) interface{} {
+			return MapFirst(itr, c.Fields()[0])
+		}, nil
 	case "last":
-		return MapLast, nil
+		return func(itr Iterator) interface{} {
+			return MapLast(itr, c.Fields()[0])
+		}, nil
+
 	case "top", "bottom":
+		// Capture information from the call that the Map function will require
+		lit, _ := c.Args[len(c.Args)-1].(*influxql.NumberLiteral)
+		limit := int(lit.Val)
+		args := topCallArgs(c)
+		fields := c.Fields()
+
 		return func(itr Iterator) interface{} {
-			return MapTopBottom(itr, c)
+			return MapTopBottom(itr, limit, args, fields, len(c.Args), c.Name)
```

Comment:
@nathanielc this is what it refactors to.  Not ideal, but should make your life easier.